python/paddle/fluid/tests/unittests/mkldnn/test_cast_mkldnn_op.py

```diff
@@ -0,0 +1,101 @@
+#   Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+from paddle.fluid.tests.unittests.op_test import OpTest, convert_float_to_uint16, convert_uint16_to_float
+import unittest
+import numpy as np
+
+import paddle
+import paddle.fluid.core as core
+import paddle.fluid as fluid
+from paddle.fluid import compiler, Program, program_guard
+
+
+@unittest.skipIf(not core.supports_bfloat16(),
+                 "place does not support BF16 evaluation")
+class TestCastBF16ToFP32MKLDNNOp(OpTest):
```

Comment:
You can even do more i.e.:
```suggestion
class TestCastBF16ToFP32MKLDNNOp(OpTest):
    def init_data(self):
        self.x = np.random.random(size=[10, 10]).astype("float32")
        self.out = convert_float_to_uint16(self.x_fp32)
    def setUp(self):
        self.init_data()
        self.inputs = {'X': self.x}
        self.outputs = {'Out': self.out}
        prepare_dtype= lambda xx: int(core.VarDesc.VarType.BF16 if xx.dtype != np.float 
                                     else core.VarDesc.VarType.FP32)
        self.attrs = {
            'in_dtype': prepare_dtype(self.x),
            'out_dtype': prepare_dtype(self.out),
            'use_mkldnn': True
        }
        self.op_type = 'cast'
..............
..............
..............
```
Then you' need to change only init_data method for later combination of tests ie.:
```suggestion
class TestCastFP32ToBF16MKLDNNOp(TestCastBF16ToFP32MKLDNNOp):
    def init_data(self):
        self.x = np.random.random(size=[2, 6]).astype("float32")
        self.out = convert_float_to_uint16(self.x_fp32)
```

Reply:
done